### DISCUSSION TITLE
Add an option instead of enable FireDBG by default

### DISCUSSION THREAD

**Discussion by dougg0k** [SeaQL/FireDBG.for.Rust#30] [Ideas]
Hi,

When you create a new rust project and you have the firedbg extension active, it automatically create a firedbg folder with target.json and version.toml

Problem is, it might not be something I want in every project or every time I create a new one.  Only when I need and want it.

I suppose, if there were an option, if you (the dev) prefer to leave it as default (the way it is), to disable that and only trigger if you open the FireDBG extension (side tab?) and choose to activate it, then it would generate the files.

It would be similar in how the debug from vscode works when you dont have the debugger yet configured for that project. You choose to configure, and a launch.json is generated in the project.

Thanks

**Comment by tyt2y3** [comment]
I agree with you. May be it should only trigger the first time your click something in the debug panel.

**Comment by tyt2y3** [comment]
If you now update the extension, you will see the following on workspaces without a `firedbg` sub-directory: 

![image](https://github.com/SeaQL/FireDBG.for.Rust/assets/1782664/8bfc3a9e-4420-4173-9bef-d593e125c492)

You'd then have to manually click "Activate", then a `firedbg` sub-directory will be created.

**Reply by dougg0k** [reply]
Nice, perhaps the FireDBG shortcut on the sidebar should be hidden till the project has the firedbg folder with the files? 

Because if there is no way to activate through there, it wont mean or do anything.

In my suggestion, I meant in add the activate button only in that firedbg sidebar tab rather than the debug tab, but obviously it's your choice.

**Reply by tyt2y3** [reply]
I don't know if that's possible in VS Code.

**Reply by dougg0k** [reply]
I see. 


I meant somewhere here and after activating, the button would vanish. 

Imo, here would be more intuitive to find instead of the run and debug tab. If the extension shortcut is to be always there.
![2024-01-08_17-23 (1)](https://github.com/SeaQL/FireDBG.for.Rust/assets/10801221/2f834fce-c2ea-44d9-a038-fcd4155baca3)

**Reply by dougg0k** [reply]
Does that make sense?

**Reply by tyt2y3** [reply]
I am sorry I don't get what you mean.

**Reply by dougg0k** [reply]
I meant that the activate button for firedbg could be more intuitive there show in the screenshot rather than where it was placed (run and debug), If the extension shortcut cannot be hidden till activated.

**Reply by tyt2y3** [reply]
Oh I get you now. It's definitely doable.
